xdsl/dialects/csl/csl_wrapper.py

```diff
@@ -0,0 +1,308 @@
+from xdsl.dialects.builtin import (
+    ArrayAttr,
+    DictionaryAttr,
+    IntegerAttr,
+    IntegerType,
+    StringAttr,
+)
+from xdsl.dialects.csl import csl
+from xdsl.dialects.utils import AbstractYieldOperation
+from xdsl.ir import Attribute, Block, Dialect, Operation, Region, SSAValue
+from xdsl.irdl import (
+    IRDLOperation,
+    Operand,
+    irdl_op_definition,
+    opt_attr_def,
+    prop_def,
+    region_def,
+    result_def,
+    traits_def,
+    var_operand_def,
+)
+from xdsl.parser import Parser
+from xdsl.printer import Printer
+from xdsl.traits import HasParent, IsTerminator, Pure
+from xdsl.utils.hints import isa
+
+
+@irdl_op_definition
+class ImportModuleOp(IRDLOperation):
+    """
+    Lightweight wrapper around `csl.import_module` that allows specifying field names directly
+    and removes the need for handling structs or setting up struct operands
+    """
```

Comment:
See it as a helper op where you don't need to do this. The idea is to not handle structs manually at this point at all. If you really want to, you can still use the underlying `csl.import_module` instead of `csl_wrapper.import_module`